### ISSUE TITLE
Spacing between icons increases when increasing vertical panel width

### ISSUE THREAD

**Issue by ghost** [mate-desktop/mate-panel#599]
#### Expected behaviour
I expected to be able to increase the width of the vertical 
panel, without affecting the spacing between icons.

#### Actual behaviour
Using a vertical panel with icons (program starters), the spacing 
between the icons gets larger when the width of the panel is 
increased, occupying much more space than needed.

#### Steps to reproduce the behaviour
Create a vertical panel. Place some program starters (icons) on it. Increase the width of the panel. Watch the spacing between the icons.

#### MATE general version
1.16.2-1

#### Package version
Debian Stable (Stretch)

#### Linux Distribution
Debian Stable (Stretch)

**Comment by monsta** [comment]
Confirmed. And I can't move the icons closer to each other when the spacing is increased - it's some invisible border. If I move too close, I just move the icon over another icon.

**Comment by lukefromdc** [comment]
I just did some testing with this, and can confirm it still present in today's (11-2-2017) git master. Not only that, it is present in all panel orientations but is more likely to be visible with the wide dimensions common on vertical panels and thus larger difference between icon size and panel size.

I fired up GtkInspector, and the icon has git css node "button" and is contained within "PanelWidget". The widget name is "ButtonWidget."  Clicking that in the GtkInspector window shows the space occupied by the button-and it comes up as a square whose dimensions equal the panel width in a vertical panel or panel height in a horizontal panel. The buttons occupy the entire square and cannot overlap. The icons within the panel are rescaled as the panel dimensions change-but the panel can be rescaled in 1px increments while the icons are normally png icons in discrete sizes in this sequence: 16, 22,24,32,48,64,and 128px square. There are also the "scalable" .svg icons, but they are also rescaled in the same steps as the .png icons, at least in the panel.

The fix for this would be to shrink the dimensions of the button widget to fit the icon, rather than expand it to fit the panel, probably by using a min(panel_width, icon_size) sort of logic when vertical or  min(panel_height, icon_size) logic when horizontal.  This would be transparent to the user unless the theme put a background on the button widget, in which case it would be plainly visible.
